ykbuild/src/completion_wrapper.rs

```diff
@@ -67,9 +67,19 @@ impl CompletionWrapper {
         p
     }
 
-    /// Returns the key and value that must be applied to the wrapped compiler's environment.
-    pub fn build_env(&self) -> (&str, &str) {
-        ("YK_COMPILER_TEMPDIR", self.tmpdir.path().to_str().unwrap())
+    /// Returns a [Vec] of `(key, value)` pairs that define environment variables necessary for the
+    /// wrapper compiler's environment.
+    pub fn build_env(&self) -> Vec<(&str, String)> {
+        vec![
+            (
+                "YK_COMPILER_TEMPDIR",
+                self.tmpdir.path().to_str().unwrap().to_owned(),
+            ),
+            (
+                "YK_COMPILER_PATH",
+                ykllvm_bin("clang").as_os_str().to_str().unwrap().to_owned(),
```

Comment:
is it correct to hardcode clang here?

I think it was OK when callsites chose their compiler, but I don't think it's right for this code to make that assumption, as this class is for both C and C++ code.

Reply:
They all chose clang (and we also found that `clang` will compile C++ code correctly, so maybe it doesn't matter?!). We can parameterise it if/when we know we need to?

Reply:
> we also found that clang will compile C++ code correctly

That surprises me! And is what confused me.

I think at the very least we should add a comment saying why this works to avoid confusion by future readers.

(Ideally I think I'd prefer the explicit selection of `clang` or `clang++`, as there are advantages to compiling C code with a C++ compiler. Some people do it to get better type handling etc -- let's not do that now)

Reply:
Good idea. Fixed in 4e60a77.

Reply:
Not quite what I meant. I was suggesting a comment that explains that `clang` will in fact compile C++ code.

Reply:
> I  was suggesting a comment that explains that clang will in fact compile C++ code

I'm not sure we entirely understand if/what the consequences of doing so might be, so I'd rather not explain things in that way.

Reply:
That was the source of my confusion you see.

I think adding:

```
// Note that we hard-code `clang` because it is quite happy to compile both C *and* C++ code!
```

Would help our future selves and others.

Reply:
I don't want to say that, because I don't know exactly how true it is. I'd rather be conservative in what I claim.

Reply:
> I don't know exactly how true it is

If we are unsure if clang correctly compiles C++ code, then we should not call clang, surely?

Reply:
I agree: I think that I should do that as the next PR.

Reply:
Thanks. Sorry to be awkward.